src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const SEPARATOR = fromFormat[fromFormat.length - 1];
+  const JOINER = toFormat[toFormat.length - 1];
+  const DATEPARTS = date.split(SEPARATOR);
+
+  while (DATEPARTS.length > 1 && DATEPARTS[DATEPARTS.length - 1] === '') {
```

Comment:
The loop that removes empty strings from the end of `DATEPARTS` is unnecessary if the input is validated correctly. If the date string is always in the correct format, there should not be any empty strings after splitting.